pandas/core/series.py

```diff
@@ -5037,12 +5041,42 @@ def between(self, left, right, inclusive=True) -> Series:
         3    False
         dtype: bool
         """
-        if inclusive:
+
+        if inclusive == "both":
             lmask = self >= left
             rmask = self <= right
-        else:
+        elif inclusive == "left":
+            lmask = self >= left
+            rmask = self < right
+        elif inclusive == "right":
+            lmask = self > left
+            rmask = self <= right
+        elif inclusive == "neither":
             lmask = self > left
             rmask = self < right
+        elif inclusive is False:
+            warnings.warn(
+                "Boolean inputs to the `inclusive` argument are deprecated in"
+                "favour of `both` or `neither`.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            lmask = self > left
+            rmask = self < right
+        elif inclusive is True:
+            warnings.warn(
+                "Boolean inputs to the `inclusive` argument are deprecated in"
+                "favour of `both` or `neither`.",
+                FutureWarning,
+                stacklevel=2,
+            )
```

Comment:
can we deduplicate this warning code.

Reply:
maybe something like...
``` 
        if inclusive is True or inclusive is False:
            warnings.warn(
                "Boolean inputs to the `inclusive` argument are deprecated in"
                "favour of `both` or `neither`.",
                FutureWarning,
                stacklevel=2,
            )
       inclusive = "both" if inclusive else "neither"
```

and then the logic you have now. On master the condition is actually based on the truthiness of `inclusive` so maybe need to maintain that behavior but the docstring clearly states `bool` so not strictly necessary.